doc/compressed_transactions.md

```diff
@@ -0,0 +1,153 @@
+| Input Count  | VarInt Flag | If 0, indicates a future VarInt. Otherwise encodes the transaction input count. |
+| Output Count | VarInt Flag | If 0, indicates a future VarInt. Otherwise encodes the transaction output count. |
+| LockTime     | 1 bit       | Wether the LockTime has a value. |
+
+### 3.4 Input Metadata
+
+| Name                        | Width       | Description |
+| ----------------------------|-------------|-------------|
+| Taproot Key or Script Spend | 1 Bits      | For Taproot: 0 for keyspend, 1 for scriptspend; for non-Taproot: 0 for p2sh, 1 for p2sh-wpkh. |
+| Standard Hash               | 1 Bits      | Boolean to determine if this Input's Signature Hash Type is standard (0x00 for Taproot, 0x01 for Legacy/Segwit). |
+| Standard Sequence           | VarInt Flag | If 0, indicates a future VarInt. Otherwise encodes the input sequence as follows: 0b01 = 0x00000000, 0b10 = 0xFFFFFFFE, 0b11 = 0xFFFFFFFF. |
+
+### 3.5.1 Output Metadata
+
+| Name                | Width  | Description |
+| --------------------|--------|-------------|
+| Encoded Script Type | 3 Bits | Encoded Script Type. |
+
+#### 3.5.2 Script Type encoding
+
+| Script Type       | Value |
+| ------------------|-------|
+| Uncompressed P2PK | 0b000 |
+| Compressed P2PK   | 0b001 |
+| P2PKH             | 0b010 |
+| P2SH              | 0b011 |
+| P2WSH             | 0b100 |
+| P2WPKH            | 0b101 |
+| P2TR              | 0b110 |
+| Unknown           | 0b111 |
+
+### 3.6 Input Data
+
+| Name                    | Width            | Description |
+| ------------------------|------------------|-------------|
+| Sequence                | Bitstream VarInt | If the Sequence was non-standard then encoded it here as a Bitstream VarInt. |
+| Hash Type               | 1 Byte           | If the Hash Type for this Input was non standard, contains a single byte. |
+| Txid Blockheight        | Bitstream VarInt | If 0 indicates a coinbase input, Otherwise contains the offset from Minimum Blockheight for the Txid. |
+| Txid/Signature Data     | Variable         | Txid/Signatures are determined to be uncompressed either by the output script of the previous transaction, or if the Txid Blockheight is zero. For each Compressed Txid/Signature See Section 3.6.1. For each Uncompressed Txid/Signature See Section 3.6.2. |
+
+### 3.6.1 Compressed Txid/Signature Data
+
+| Name              | Width            | Description |
+| ------------------|------------------|-------------|
+| Txid Block Index  | Bitstream VarInt | Contains the flattend index from the Txid Blockheight for the Vout. |
+| Signature         | 64 Bytes         | Contains the 64 byte signature. |
+
+### 3.6.2 Uncompressed Txid/Signature Data
+
+| Name                    | Width         | Description |
+| ------------------------|---------------|-------------|
+| Txid                    | 32 Bytes      | Contains the 32 byte Txid. |
```

Comment:
In ddde5c7defae80058c35bcd5fdc27ee57935495f:

In the uncompressed case we still need an output index (within the transaction). We should probably use a VarInt for this.

Reply:
Fixed, Thanks!